src/CoreWCF.Http/tests/DependencyInjection/ServiceTypeErrorMessageTests.cs

```diff
@@ -0,0 +1,156 @@
+                ITestService channel = factory.CreateChannel();
+                
+                // Assert - Making a request should throw a FaultException with the service type name
+                var exception = Assert.Throws<System.ServiceModel.FaultException<System.ServiceModel.ExceptionDetail>>(() => channel.GetData("test"));
+                
+                // Verify that the error message contains the full type name
+                Assert.Contains(typeof(SingletonServiceWithNoDefaultConstructor).FullName, exception.Detail.Message);
+            }
+        }
+
+        private class StartupWithProblematicService
+        {
+            public void ConfigureServices(IServiceCollection services)
+            {
+                // Note: We're NOT registering ServiceWithNoDefaultConstructor in DI
+                // This will cause the error when the service is tried to be instantiated
+                services.AddServiceModelServices();
+            }
+
+            public void Configure(IApplicationBuilder app)
+            {
+                app.UseServiceModel(builder =>
+                {
+                    // This will fail because ServiceWithNoDefaultConstructor is not in DI and has no default constructor
+                    // Using ServiceOptions to enable exception detail in faults so we can verify the error message
+                    builder.AddService<ServiceWithNoDefaultConstructor>(options =>
+                    {
+                        options.DebugBehavior.IncludeExceptionDetailInFaults = true;
+                    });
+                    builder.AddServiceEndpoint<ServiceWithNoDefaultConstructor, ITestService>(new BasicHttpBinding(), "/testservice");
+                });
+            }
+        }
+
+        [System.ServiceModel.ServiceContract]
+        internal interface ITestService
+        {
+            [System.ServiceModel.OperationContract]
+            string GetData(string value);
+        }
+
+        private class StartupWithProblematicSingletonService
+        {
+            public void ConfigureServices(IServiceCollection services)
+            {
+                // Note: We're NOT registering SingletonServiceWithNoDefaultConstructor in DI
+                // This will cause the error when the service is tried to be instantiated
+                services.AddServiceModelServices();
+            }
+
+            public void Configure(IApplicationBuilder app)
+            {
+                app.UseServiceModel(builder =>
+                {
+                    // This will fail because SingletonServiceWithNoDefaultConstructor is not in DI and has no default constructor
+                    // Using ServiceOptions to enable exception detail in faults so we can verify the error message
+                    builder.AddService<SingletonServiceWithNoDefaultConstructor>(options =>
+                    {
+                        options.DebugBehavior.IncludeExceptionDetailInFaults = true;
```

Comment:
There is no need to set IncludeExceptionDetailsInFaults when the service is a Singleton as the exception isn't exposed through a service call.